### PR TITLE
Add new ERIO test type.

### LINKED ISSUE #306: Need testcase to replace ERS_IOP
Going to call it IOP.

### PR DESCRIPTION
**Pull request by jgfouca**
Tests that different PIO methods are compatible

Add ERIO test to cime_developer

Test suite: ERIO.f45_g37.X
Test baseline: 
Test namelist changes: 
Test status: bit for bit

Fixes #306

User interface changes?: New ERIO test type

Code review: jedwards

### CODE REVIEW

**Comment by jgfouca** [comment]
@jedwards4b before merging this, I was wondering if you could help me test it. There's a couple things going on that concern me.

First, I cannot tell by looking at the sharedlibs if multiple PIO are actually being built.

Second, on melvin, it appears that PIO_TYPENAMES only include netcdf and pnetcdf even though there are several others in the valid list. I'm wondering why.

File: utils/python/CIME/SystemTests/erio.py [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+
+class ERIO(SystemTestsCommon):
+
+    def __init__(self, case):
+        """
+        initialize an object interface to file env_test.xml in the case directory
+        """
+        SystemTestsCommon.__init__(self, case, expected=["TEST"])
+
+        self._pio_types = self._case.get_env("run").get_valid_values("PIO_TYPENAME")
```
Comment (jgfouca) [inline utils/python/CIME/SystemTests/erio.py]: Not sure if this is really giving me all the possible pio types.

File: utils/python/CIME/SystemTests/erio.py [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+
+        for pio_type in self._pio_types:
+            if pio_type == "default":
+                continue
+            else:
+                self._case.set_value("PIO_TYPENAME", pio_type)
+                self._case.flush()
+
+                self.clean_build()
+                self.build_indv(sharedlib_only=sharedlib_only, model_only=model_only)
```
Comment (jgfouca) [inline utils/python/CIME/SystemTests/erio.py]: Not sure if PIO if the case.set_value above is having any impact on the PIO build here.

File: utils/python/CIME/XML/entry_id.py [hunk 3]
```diff
@@ -140,6 +140,17 @@ def _get_description (self, node):
     def _get_group (self, node):
         return self._get_node_element_info(node, "group")
 
+    def get_valid_values(self, vid):
+        node = self.get_optional_node("entry", {"id":vid})
+        if node is None:
+            return None
+
+        vv_node = self.get_optional_node("valid_values", root=node)
```
Comment (jedwards4b) [inline utils/python/CIME/XML/entry_id.py]: The code in lines 148-152 is pretty widely used.  I was about to tell you that I've added code in generic_xml.py that does this, but then I realised that that code is still on a branch...

**Pull request merged**